ports/qemu/mphalport.c

```diff
@@ -64,3 +69,29 @@ mp_uint_t mp_hal_stdout_tx_strn(const char *str, size_t len) {
     #endif
     return len;
 }
+
+mp_uint_t mp_hal_ticks_ms(void) {
+    extern mp_uint_t systick_ticks_ms(void);
```

Comment:
Please move this declaration, and the `systick_ticks_us()` below, to the top of this file, just before `mp_hal_stdio_poll()` (and the `extern` is not needed).

Alternatively, could put them in `mcu/systick.h`.